src/main/java/com/teragrep/net_01/channel/context/IngressImpl.java

```diff
@@ -250,4 +259,46 @@ private void activateBuffers(List<BufferLease> bufferLeases) {
     public AtomicBoolean needWrite() {
         return needWrite;
     }
+
+    @Override
+    public void register(Clock clock) {
+        lock.lock();
+        try {
+            if (!interestedClocks.isEmpty()) {
+                throw new IllegalStateException("only one interested clock is allowed");
+            }
+            interestedClocks.add(clock);
+            establishedContext.interestOps().add(OP_READ);
+        }
+        finally {
+            lock.unlock();
+        }
+    }
+
+    @Override
+    public void unregister(Clock clock) {
+        lock.lock();
+        try {
+            if (!interestedClocks.contains(clock)) {
+                throw new IllegalStateException("clock not registered");
```

Comment:
Maybe change the formatting to `Unable to unregister ingress clock, it is not registered`